packages/core/src/interpreter.ts

```diff
@@ -131,158 +118,131 @@ export class Interpreter<
   public _parent?: ActorRef<any>;
   public name: string;
   public ref: ActorRef<TEvent>;
+  private _actorContext: ActorContext<TEvent, SnapshotFrom<TBehavior>>;
 
   /**
    * The globally unique process ID for this invocation.
    */
   public sessionId: string;
-  private forwardTo: Set<string> = new Set();
+
+  // TODO: remove
+  public _forwardTo: Set<string> = new Set();
 
   /**
    * Creates a new Interpreter instance (i.e., service) for the given machine with the provided options, if any.
    *
-   * @param machine The machine to be interpreted
+   * @param behavior The machine to be interpreted
    * @param options Interpreter options
    */
-  constructor(
-    public machine: StateMachine<
-      TContext,
-      TEvent,
-      any,
-      any,
-      TResolvedTypesMeta
-    >,
-    options?: InterpreterOptions
-  ) {
+  constructor(public behavior: TBehavior, options?: InterpreterOptions) {
     const resolvedOptions = {
       ...defaultOptions,
       ...options
     } as Required<InterpreterOptions>;
 
     const { clock, logger, parent, id } = resolvedOptions;
 
-    const resolvedId = id !== undefined ? id : machine.key;
-
-    this.name = this.id = resolvedId;
+    this.name = this.id = id;
     this.logger = logger;
     this.clock = clock;
     this._parent = parent;
-
     this.options = resolvedOptions;
-
     this.ref = this;
-
-    this.sessionId = this.ref.name;
+    // TODO: this should come from a "system"
+    this.sessionId = registry.bookId();
+    this._actorContext = {
+      self: this,
+      name: this.id ?? 'todo',
+      sessionId: this.sessionId,
+      logger: this.logger,
+      exec: (fn) => {
+        fn();
+      },
```

Comment:
This isn't used anywhere in the codebase as far as I can see. Do we need it?

Reply:
Something like this will be in the future, so e.g. we can override execution with a no-op or similar (useful for simulation), but we can remove this for now. Just need to keep a mental note that `execAction(...)` is serving this role right now,.